app.js

```diff
@@ -1,68 +1,45 @@
+import { app, errorHandler } from "mu";
+import fs from "fs";
+import { FILE_JSONAPI_TYPE } from "./cfg";
+import { getFile, setFileSource, storeFile } from "./lib/file";
+import { execWithRetry } from "./lib/util";
+import { deleteFile, downloadPdf, uploadFile } from "./lib/graph-api";
 
-app.post('/files/:id/convert', async (req, res) => {
+app.post("/files/:id/convert", async (req, res, next) => {
   const fileId = req.params.id;
   const file = await getFile(fileId);
-  file.path = file.physicalUri.replace('share://', '/share/');
+  file.path = file.physicalUri.replace("share://", "/share/");
 
   try {
-    const uploadSessionPayload = {
-      item: {
-        '@microsoft.graph.conflictBehavior': 'rename',
-      },
-    }
-    const uploadSession = await LargeFileUploadTask.createUploadSession(
-      client,
-      `/sites/${SITE_ID}/drive/root:/${file.id}.${file.extension}:/createUploadSession`,
-      uploadSessionPayload,
-    );
     const stats = fs.statSync(file.path);
-    const fileSize = stats.size;
+    file.size = stats.size;
     const readStream = fs.createReadStream(file.path);
-    const fileObject = new StreamUpload(readStream, file.name, fileSize);
-    const options = {
-      rangeSize: 327680,
-      uploadEventHandlers: {
-        progress: (range) => {
-          console.info(
-            `Progress uploading file <${file.uri}>. Bytes: [${range.minValue}, ${range.maxValue}]`
-          )
-        }
-      },
-    };
-    const uploadTask = new LargeFileUploadTask(client, fileObject, uploadSession, options);
-    await uploadTask.upload();
-  } catch (err) {
-    console.log(`Error uploading file: ${JSON.stringify(err)}`);
-    console.error(err);
-  }
 
-  const buffer = await client
-        .api(`/sites/${SITE_ID}/drive/root:/${file.id}.${file.extension}:/content?format=pdf`)
-        .get();
+    await execWithRetry(() => uploadFile(file, readStream));
+
+    const buffer = await execWithRetry(() => downloadPdf(file));
 
-  const newFile = await storeFile(`${file.id}.pdf`, buffer, file.isDraftFile);
-  await setFileSource(file.uri, newFile.uri);
+    const newFile = await storeFile(`${file.id}.pdf`, buffer, file.isDraftFile);
+    await setFileSource(file.uri, newFile.uri);
 
-  await client
-    .api(`/sites/${SITE_ID}/drive/root:/${file.id}.${file.extension}`)
-    .delete();
+    await execWithRetry(() => deleteFile(file));
 
-  return res
-    .status(200)
-    .send({
-      data: [{
-        type: FILE_JSONAPI_TYPE,
-        id: newFile.id,
-        attributes: {
-          uri: newFile.uri,
-        }
-      }]
+    return res.status(200).send({
+      data: [
+        {
+          type: FILE_JSONAPI_TYPE,
+          id: newFile.id,
+          attributes: {
+            uri: newFile.uri,
+          },
+        },
+      ],
     });
+  } catch (err) {
+    console.log(`Error converting DOCX file: ${JSON.stringify(err)}`);
+    console.log(err);
```

Comment:
I remember you saying something about logged objects not getting flagged for garbage collection.
Should we just log the error stringified?

Reply:
reading about `console.log(err.stack)` or `console.trace(err)` not sure about garbage collection in those cases

Reply:
1 = log(err.stack)
2 = trace(err)
![image](https://github.com/user-attachments/assets/aef92dc6-da61-4852-8ffd-7f9135b09045)

Reply:
how it looks in chrome inspector
![image](https://github.com/user-attachments/assets/056212cb-fd8c-4589-8544-55f78f96f7f9)

Reply:
getting side tracked here :sweat_smile:

Reply:
No, you're right. That `console.log(err)` should go. Thanks for pointing that out.